src/services/user/utils/aggregations.js

```diff
@@ -397,6 +436,10 @@ const createMultiDocumentAggregation = ({
 		stageSort(aggregation, sort);
 	}
 
+	if (searchQuery) {
+		stageSearch(aggregation, searchQuery);
+	}
```

Comment:
code looks nice, but this one should be done earlier. This filters out a lot of entries and it do something simular as match, so do it before or after the match. You have to think about every stage a query doing staff on the data from the stage before, so for example constentStatus and classes are queries taking more effort, so let them do as less as posible.